hexrd/WPPF.py

```diff
@@ -4899,6 +4859,148 @@ def scale(self, value):
+    ubs.append(1e3)
+    ubs.append(1.)
+    varies = 7*[False]
+
+    params.add_many(names, values=values,
+                    varies=varies, lbs=lbs, ubs=ubs)
+
+    if isinstance(mat, Phases_Rietveld):
+        """
+        phase file
+        """
+        for p in mat:
+            m = mat[p]
+            _add_atominfo_to_params(params, m)
+
+    elif isinstance(mat, Material):
+        """
+        just an instance of Materials class
+        this part initializes the lattice parameters in the
+        """
+        _add_atominfo_to_params(params, mat)
+
+    elif isinstance(mat, list):
+        """
+        a list of materials class
+        """
+        for m in mat:
+            _add_atominfo_to_params(params, m)
+
+    elif isinstance(mat, dict):
+        """
+        dictionary of materials class
+        """
+        for k, m in mat.items():
+            _add_atominfo_to_params(params, m)
+
+    else:
+        msg = f"_generate_default_parameters: "
+        "incorrect argument. only list, dict or "
+        "Material is accpeted."
+        raise ValueError(msg)
+
+    return params
+
+
+def _add_atominfo_to_params(params, mat):
+    """
+    03/12/2021 SS 1.0 original
+    given a material, add the required
+    lattice parameters, atom positions,
+    occupancy, DW factors etc.
+    """
+
+    lp = np.array(mat.lparms)
+    rid = list(_rqpDict[mat.latticeType][0])
+
+    lp = lp[rid]
+    name = _lpname[rid]
+
+    phase_name = mat.name
+
+    for n, l in zip(name, lp):
+        nn = f"{phase_name}_{n}"
+        """
+        is l is small, it is one of the length units
+        else it is an angle
+        """
+        if(n in ['a', 'b', 'c']):
+            params.add(nn, value=l, lb=l-0.05,
+                       ub=l+0.05, vary=False)
+        else:
+            params.add(nn, value=l, lb=l-1.,
+                       ub=l+1., vary=False)
+
+    atom_pos = mat.atom_pos[:, 0:3]
+    occ = mat.atom_pos[:, 3]
+    atom_type = mat.atom_type
+
+    atom_label = _getnumber(atom_type)
+
+    for i in range(atom_type.shape[0]):
+
+        Z = atom_type[i]
+        elem = constants.ptableinverse[Z]
+
+        nn = f"{phase_name}_{elem}{atom_label[i]}_x"
+        params.add(
+            nn, value=atom_pos[i, 0],
+            lb=0.0, ub=1.0,
+            vary=False)
+
+        nn = f"{phase_name}_{elem}{atom_label[i]}_y"
+        params.add(
+            nn, value=atom_pos[i, 1],
+            lb=0.0, ub=1.0,
+            vary=False)
+
+        nn = f"{phase_name}_{elem}{atom_label[i]}_z"
+        params.add(
+            nn, value=atom_pos[i, 2],
+            lb=0.0, ub=1.0,
+            vary=False)
+
+        nn = f"{phase_name}_{elem}{atom_label[i]}_occ"
+        params.add(nn, value=occ[i],
+                   lb=0.0, ub=1.0,
+                   vary=False)
+
+        if(mat.aniU):
+            U = mat.U
+            for j in range(6):
+                nn = f"{phase_name}_{elem}{atom_label[i]}"
+                "_{nameU[j]}"
```

Comment:
Add parentheses and `f` at the beginning of the second line.

Reply:
Fixed.